### PR TITLE
HWP-271 Create test file for GET /api/community

### PR DESCRIPTION
**Pull request by mattsiel**
# Description

This PR includes the following proposed change(s):

- creates a jest configuration file for test files to run
- creates tests to get communities from the api endpoint
- tests different scenarios, such as whether the user is logged in or not.

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/tests/api-getCommunity.test.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  });
+
+  test('API returns description "Forbidden."', async () => {
+    let response = await request
+      .get('/community')
+      .set({authorization: `Bearer ${token}11`})
+
+    expect(response.text).toBe("Forbidden.");
+  });
+});
```
Comment (mattsiel) [inline app/server/tests/api-getCommunity.test.js:135]:
```suggestion
});

```

File: app/server/tests/api-getCommunity.test.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
+require('dotenv').config();
+const supertest = require('supertest');
+
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:3]: lets tighten up the whitespace at the top of this file 😄

File: app/server/tests/api-getCommunity.test.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    test('API returns description "Not Authorized."', async () => {
+      let response = await request
+        .get('/community')
+
+      expect(response.text).toBe("Not Authorized.");
+    });
+});
+
+
+
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:37]:
```suggestion

```

File: app/server/tests/api-getCommunity.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            "password": "Test123!"
+        });
+
+      token = response.body.token;
+      expect(response.status).toBe(201);
+  });
+});
+
+
+
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:54]:
```suggestion

```

File: app/server/tests/api-getCommunity.test.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    let response = await request
+      .get('/community')
+      .set({authorization: `Bearer ${token}`})
+
+    expect(" " + response.text+ " ").toContain(welcomeCommunityDescript);
+    expect(" " + response.text + " ").toContain(welcomeCommunityTitle);
+  });
+});
+
+
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:83]:
```suggestion

```

File: app/server/tests/api-getCommunity.test.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  test('API returns description "Not Authorized."', async () => {
+    let response = await request
+      .get('/community')
+
+    expect(response.text).toBe("Not Authorized.");
+  });
+});
+
+
+
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:109]:
```suggestion

```

File: app/server/tests/api-getCommunity.test.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      token = response.body.token;
+      expect(response.status).toBe(201);
+  });
+});
+
+
+
+describe('Testing the get communities function after logging in', () => {
+
+  test('API returns successful response', async () => {
```
Comment (akroon3r) [inline app/server/tests/api-getCommunity.test.js:57]: I feel like the test that starts on 57 and 65 to be very similar. Can you explain what the difference is?

Comment (mattsiel) [reply app/server/tests/api-getCommunity.test.js:51]: like incase it gave a 201, it would get caught there.

**Review by dbarkowsky — APPROVED** [review]

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**